src/proto/h1/decode.rs

```diff
@@ -19,8 +19,11 @@ use self::Kind::{Chunked, Eof, Length};
 /// This limit is currentlty applied for the entire body, not per chunk.
 const CHUNKED_EXTENSIONS_LIMIT: u64 = 1024 * 16;
 
-/// Maximum number of trailers allowed in a chunked body.
-const TRAILERS_LIMIT: u64 = 1024;
+/// Maximum number of trailer fields allowed in a chunked body.
+const TRAILERS_FIELD_LIMIT: u64 = 1024;
+
+/// Maximum number of bytes allowed for all trailer fields.
+const TRAILER_LIMIT: u64 = TRAILERS_FIELD_LIMIT * 64;
```

Comment:
I wonder if we want to expose an option to the user for this.

Reply:
sure. when i think about the use cases for this:

- the client sends `TE: trailers` and wants to limit the amount of trailers to read in the server's response
- the client sends a chunked body with trailer fields and the server wants to limit the amount of trailers to read in the client's request

Looking at prior art:

- We can include `h1_max_trailer_size` and `h1_max_trailer_fields` in `client::conn::http1::Builder`
- We can include `h1_max_trailer_size` and `h1_max_trailer_fields` in `server::conn::http1::Builder`

Reply:
Let's add this in this PR too, I think it's pretty simple

Reply:
Let's add this in this PR too, I think it's simple enough to include this

Reply:
I've been thinking of adding a `max_header_size()` for the HTTP/1 builder, to mirror the HTTP/2 side. Does it make sense for that to be used by both headers and trailers? Or do people need to allow different sizes?

Reply:
Most uses of trailer fields, that I have observed, add 1-2 headers that are relatively short. I don't think people would want more capacity than what was for `max_header_size()`. People may want to set a lower capacity for trailers than what `max_header_size()` allows, but I am unsure how common this would be.

I think we should use `max_header_size()` to enforce both for now. If there is a real use case that people bring up, we can always add `max_trailers()` and `max_trailers_size()` in the future.

If we want to go this route, then I can update the trailer parsing logic to check against `max_headers()`. In a future PR, I can:

- Add `max_header_size()` to the client and server builders.
- Update the trailer parsing logic to check against `max_headers_size()`
- Update the header parsing logic `src/proto/h1/role.rs` to enforce `max_headers_size()`

Reply:
i added 6c01c879e9f62a543e8865e536dd7d1561619a1f to resolve this